src/static/bitbucket/components/App.tsx

```diff
@@ -71,10 +73,19 @@ const App = () => {
   let refreshTimeoutId: Timeout;
 
   const pollAbleToLand = () => {
-    const isTabInForeground = !document.hidden;
-    let refreshIntervalMs = inViewRef.current ? 10000 : 20000;
-
-    const checkPromise = isTabInForeground ? checkIfAbleToLand() : Promise.resolve();
+    const widgetSettings = getWidgetSettings();
+    const isVisible =
+      !document.hidden && (widgetSettings.refreshOnlyWhenInViewport ? inViewRef.current : true);
+
+    // If only refreshing when in viewport, uses `refreshInterval`,
+    // Else uses `refreshInterval` when in viewport and twice the timeout when not in viewport
+    let refreshIntervalMs = widgetSettings.refreshOnlyWhenInViewport
+      ? widgetSettings.refreshInterval
+      : inViewRef.current
+      ? widgetSettings.refreshInterval
+      : widgetSettings.refreshInterval * 2;
```

Comment:
nit: just hard to read otherwise
```suggestion
    let refreshIntervalMs = widgetSettings.refreshOnlyWhenInViewport
      ? widgetSettings.refreshInterval
      : inViewRef.current
        ? widgetSettings.refreshInterval
        : widgetSettings.refreshInterval * 2;
```

Reply:
I tried to do the same earlier, but prettier wants to back to the version without indentation. Let me break try to break it up.